components/gmServiceGmail.js

```diff
@@ -233,10 +240,17 @@ gmServiceGmail.prototype = {
   _cookieLoader: function(aCookies) {
     this._log("Start the cookie loader...");
 
-    aCookies.forEach(function(cookie, index, array) {
+    aCookies.forEach(function(cookie) {
       this._log("cookie name = " + cookie.name);
       this._log("cookie value = " + cookie.value);
-      this._cookieManager.add(cookie.host, cookie.path, cookie.name, cookie.value, cookie.isSecure, cookie.isHttpOnly, cookie.isSession, cookie.expires);
+
+      // If cookie.expires equals to 0 then the cookie should be set as a session cookie. It doesn't happen so need to set expiration time manually.
+      // Expiration time is in seconds from epoch, now will be 60 seconds from now
+      var cookieExpirationTime = !cookie.expires
```

Comment:
My bad, missed that I'm changing `cookie.expires`. Will fix to use only the new variable. 

Shouldn't change function's argument.